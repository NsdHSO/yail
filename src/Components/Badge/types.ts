export interface IBadgeProps {
  title: string;
  className?: string;
}
