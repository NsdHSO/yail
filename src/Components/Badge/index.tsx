import classNames from 'classnames';
import { FC } from 'react';

interface IBadgeProps {
  title: string;
  className?: string;
}

const Badge: FC<IBadgeProps> = ({ title, className }) => {
  return (
    <span
      className={classNames(
        'border border-primary-text-color text-primary-text-color rounded px-1 pb-[2px] pt-[1px] text-[10px] relative top-[-2px]',
        className
      )}
    >
      {title}
    </span>
  );
};

export default Badge;
