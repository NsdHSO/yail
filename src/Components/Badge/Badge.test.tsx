import { render, screen } from "@testing-library/react";

import Badge from "./";

describe("Badge component", () => {
  const testTitle = "Test Badge";

  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(<Badge title={testTitle} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the title correctly", () => {
    render(<Badge title={testTitle} />);
    const badgeElement = screen.getByText(testTitle);
    expect(badgeElement).toBeInTheDocument();
  });

  test("applies custom class name", () => {
    const customClass = "custom-class";
    render(<Badge title={testTitle} className={customClass} />);
    const badgeElement = screen.getByText(testTitle);
    expect(badgeElement).toHaveClass(customClass);
  });
});
