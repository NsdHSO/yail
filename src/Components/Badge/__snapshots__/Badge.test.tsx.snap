// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <span
    class="relative top-[-2px] rounded border border-primary-text-color px-1 pb-[2px] pt-[1px] text-[10px] text-primary-text-color"
  >
    Test Badge
  </span>
</DocumentFragment>
`;
